### ISSUE TITLE
100% reproducible crash in simplest basic application [Android only]

### ISSUE THREAD

**Issue by Omelyan** [software-mansion/react-native-reanimated#2905]
## Description
About two months ago 100% recreating crash was issued #2722, in the simplest and fundamental case, even an PR was made #2853, still no reaction...
Here it is:

### Actual behavior & steps to reproduce
"npx react-native init test"
install react-native-reanimated [(installation for Android)](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/#android)
replace "App.js" by the following code and run "npx react-native run-android --variant=release"

## Snack or minimal code example
```
import React, {useEffect} from 'react';

import Reanimated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

const App = () => {
  const sv = useSharedValue(0);

  const rs = useAnimatedStyle(() => ({
    transform: [{translateY: sv.value}],
  }));

  useEffect(() => {
    sv.value = withTiming(500, {duration: 3000});
  }, [sv]);

  return (
    <Reanimated.View
      style={[{height: 100, aspectRatio: 1, backgroundColor: 'tomato'}, rs]}
    />
  );
};

export default App;
```
It will crash if run in release mode (API 29 or below). But if you run it in dev. mode, then nope.
Also, it won't crash if you replace ```transform: [{translateY: sv.value}]``` with ```transform: [{translateY: 0}]``` or add a delay before rendering the styled element.

## Package versions
The latest.
Should note that this bug appeared in the latest 2.3.x release and in the previous one everything was fine.

## Affected platforms
- [x ] Android
- [ ] iOS
- [ ] Web

**Comment by kmagiera** [comment]
Hi @Omelyan – thank you for reporting the issue. It does not seem, however, like this can be reproduced using the steps you provided. The issue and PR you link to fixes the use-case of animating third-party component properties and we have not been able to reproduce this behavior for things like regular view's transforms (they are always of the correct type).

In any case, it would be easier if you'd provided the versions you are using as the issue seem to be specific to your setup – either the version of packages you are using or the type of device you are running the test on. For comparison the versions I tested your app with are:
```
"react": "17.0.2",
"react-native": "0.67.2",
"react-native-reanimated": "^2.3.1
```
I run the test on Android SDK 30 emulator.

**Comment by joy-betterhalf** [comment]
@kmagiera This issue is happening to us as well and it seems like not happening on every device. @Omelyan It's better you provide more information like you face this crash on Emulator or Real device? Also if a Real device what's the device you are using?

**Comment by Omelyan** [comment]
@kmagiera @joy-betterhalf  I've checked this in various cases and now can say that it's recreatable on API 29 or below. + Reanimated library was updated two days ago, so the PR mentioned before was implemented and now the issue seems to be fixed.

The issue is reproducible with the reanimated library that was before this commit: https://github.com/software-mansion/react-native-reanimated/commit/ca620a3790049739479c081fd0ecc151612359b4, the app (see code in original post) crashes immediately (if running in release mode) on Android 10 (API 29 or below).

After changes in recent commits, issue seems to be the resolved.
